lang/cs/Org.Apache.REEF.Tang/Exceptions/ClassHierarchyException.cs

```diff
@@ -19,7 +19,7 @@
 
 namespace Org.Apache.REEF.Tang.Exceptions
 {
-    public sealed class ClassHierarchyException : SystemException
+    public sealed class ClassHierarchyException : Exception
```

Comment:
Most of the exceptions we define in REEF.NET inherit directly from Exception. Therefore, I think we should also have ClassHierarchyException inherit directly from Exception, just like o.a.r.Tang.Exceptions.{IllegalStateException,InjectionException,TangApplicationException}. If we inherit from ReefRuntimeException, that makes the class hierarchy unnecessarily deep, which is undesirable, unless there is a strong reason for doing it.